### PR TITLE
Fix update_deps to grab correct platform digest, rerun

### PR DESCRIPTION
**Pull request by dhalperi**
Suggest reviewing commit by commit.

In #1655, I ran into issues where `update_deps.sh` provided invalid digests. My belief is that
distroless started shipping multi-platform images (`linux/amd64` and `linux/arm64`), so `crane`
was grabbing the multi-platform manifest instead of the `linux/amd64` digest. In the first two
commits, I

1. upgrade external repos to latest version of `crane`
2. add `linux/amd64` platform as default to `update_deps` binary

Then I

3. ran the new `update_deps.sh`, and it produced valid digests.

~Open Q: presumably, rules_docker will want to support arm images at some point? Future work?~ Answered my question here: https://github.com/bazelbuild/rules_docker/issues/1155#issuecomment-571149145

---
This change is [Reviewable](https://reviewable.io/reviews/bazelbuild/rules_docker/1666)

### CODE REVIEW

**Comment by dhalperi** [comment]
hi @smukherj1  - a followup to the PR we worked on last week, but this time the tests are fixed. I uncovered why the `update_deps.sh` script was producing bad data.

**Comment by dhalperi** [comment]
@smukherj1 - anything blocking review here? AFAICT every test that can be run has passed.

**Comment by smukherj1** [comment]
@dhalperi Apologies for the delay. Overall, rules_docker ownership/maintenance staffing is currently not enough to keep up with the incoming volume of PRs & issues. We also don't have a good system to track the state of various PRs/issues such as those ready for review or awaiting action from the maintainers vs need action from the PR author.

We're aware of this issue and actively exploring ways to improve this. However, in the meantime PRs may not necessarily get responded to or face delays.

TL; DR there was nothing blocking a review for your PR. Our maintainers just never got around to it or might have missed it.

**Review by smukherj1 — APPROVED** [review]
